redis/asyncio/connection.py

```diff
@@ -718,7 +720,14 @@ async def on_connect(self) -> None:
             await self.send_command("CLIENT", "SETNAME", self.client_name)
             if str_if_bytes(await self.read_response()) != "OK":
                 raise ConnectionError("Error setting client name")
-
+        if self.is_connected is False:
+            try:
+                await self.send_command("CLIENT", "SETINFO", "LIB-NAME", "redis-py")
+                await self.read_response()
+                await self.send_command("CLIENT", "SETINFO", "LIB-VERSION", get_lib())
+                await self.read_response()
+            except Exception:
```

Comment:
I think it is better to catch more specific exception, maybe ResponseError, WDYT?